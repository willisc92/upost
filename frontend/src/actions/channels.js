import API from "../utils/API";

// SET_CHANNELS
export const setChannels = (channels) => ({
    type: "SET_CHANNELS",
    channels
});

// START_SET_CHANNELS
export const startSetChannels = () => {
    return (dispatch) => {
        API.get("channels/", {
            params: {
                user: localStorage.getItem("user_id")
            }
        }).then((result) => {
            dispatch(setChannels(result.data));
        });
    };
};

export const addChannel = (channel) => {
    return (dispatch) => {
        API.post("channels/", {
            ...channel,
            user: localStorage.getItem("user_id")
        })
            .then((result) => {
                dispatch(startSetChannels());
            })
            .catch((err) => {
                alert(err);
            });
    };
};
