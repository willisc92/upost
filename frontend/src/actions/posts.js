import API, { setContentToForm, resetContentType } from "../utils/API";
import { getCurrentUser } from "./auth";
import moment from "moment";

/**
 * POST_START.
 * action generator.
 * Resets the error and sets loading status.
 *
 * @returns {Object} object to reset error and loading status
 */
export const postStart = () => ({
    type: "POST_START"
});

/**
 * SET_POSTS.
 * action generator.
 * adds user's posts to store.
 *
 * @param {Object[]} posts posts to add to store
 * @returns {Object} object to add posts to store
 */
export const setPosts = (posts) => ({
    type: "SET_POSTS",
    posts
});

/**
 * START_SET_POSTS.
 * from API gets the current user's posts and adds to store.
 *
 * @returns dispatch function
 */
export const startSetMyPosts = () => {
    return (dispatch) => {
        dispatch(postStart());
        getCurrentUser()
            .then((res) => {
                API.get("posts/", {
                    params: {
                        user: res.data.id
                    }
                })
                    .then((result) => {
                        dispatch(postSuccess());
                        dispatch(setPosts(result.data));
                    })
                    .catch((err) => {
                        dispatch(postFail(err));
                    });
            })
            .catch((err) => {
                console.log(JSON.stringify(err));
            });
    };
};

/**
 * SET_INTEREST_RANDOM_POSTS.
 * action generator.
 * resets the interestRandomPosts in store to be an empty array.
 *
 * @returns {Object} object to reset interestRandomPosts
 */
export const setInterestRandomPosts = () => {
    return { type: "SET_INTEREST_RANDOM_POSTS" };
};

/**
 * ADD_INTEREST_RANDOM_POSTS.
 * action generator.
 * adds a new interest and associated random posts to store.
 *
 * @param {Object} interestPosts contains tag and an array of posts
 * @returns {Object} object to add interest and posts to interestRandomPosts
 */
export const addInterestRandomPosts = (interestPosts) => {
    return { type: "ADD_INTEREST_RANDOM_POSTS", interestPosts };
};

/**
 * START_SET_INTEREST_RANDOM_POSTS.
 * resets the interestRandomPosts store.
 * for each interest provided gets random posts from API and adds to store.
 *
 * @param {Object[]} interests an array of interest objects or interest tags
 * @returns {Function} dispatch function
 */
export const startSetInterestRandomPosts = (interests) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            dispatch(setInterestRandomPosts());
            // if we have a list of interest objects
            if (interests.length > 0 && !!interests[0].interest_tag) {
                for (let i = 0; i < interests.length; i++) {
                    API.get("random-posts/", {
                        params: {
                            interest: interests[i].interest_tag
                        }
                    }).then((result) => {
                        if (result.data.length > 0) {
                            dispatch(addInterestRandomPosts({ tag: interests[i].interest_tag, posts: result.data }));
                        }
                    });
                }
            }
            // if we have a list of interest tags
            else {
                for (let i = 0; i < interests.length; i++) {
                    API.get("random-posts/", {
                        params: {
                            interest: interests[i]
                        }
                    }).then((result) => {
                        if (result.data.length > 0) {
                            dispatch(addInterestRandomPosts({ tag: interests[i], posts: result.data }));
                        }
                    });
                }
            }
            dispatch(postSuccess());
            resolve(true);
        });
    };
};

/**
 * START_GET_POST.
 * from API gets a specifc post and adds to store.
 *
 * @param {number} id post id to get
 * @returns {Promise}
 */
export const startGetPost = (id) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            API.get("posts/", {
                params: {
                    post_id: id
                }
            })
                .then((result) => {
                    dispatch(postSuccess());
                    dispatch(setPosts(result.data));
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * GET_POST.
 * from API gets a specific post, but does not add to store.  Does not dispatch any actions.
 *
 * @param {number} id post id to get
 */
export const getPost = (id) => {
    return new Promise((resolve, reject) => {
        API.get("posts/", {
            params: {
                post_id: id
            }
        })
            .then((result) => {
                resolve(result.data);
            })
            .catch((err) => {
                reject(err);
            });
    });
};

/**
 * ADD_POST.
 * sends post request to API to add a new user created post.
 * upon sucessful post reloads user posts.
 *
 * @param {Object} post post object to add to API
 * @returns {Promise}
 */
export const addPost = (post) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            if (post.toString() == "[object FormData]") {
                setContentToForm();
            }
            dispatch(postStart());
            API.post(`posts/`, post)
                .then((result) => {
                    if (post.toString() == "[object FormData]") {
                        resetContentType();
                    }
                    dispatch(postSuccess());
                    resolve(result);
                })
                .catch((err) => {
                    resetContentType();
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * EDIT_POST.
 * sends put request to API to edit a user's post.
 * upon sucessful post reloads user posts.
 *
 * @param {number} id post id to edit
 * @param {Object} updates updates to be made to the post
 */
export const editPost = (id, updates) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            if (updates.toString() == "[object FormData]") {
                setContentToForm();
            }
            dispatch(postStart());
            API.patch(`posts/${id}/`, updates)
                .then((result) => {
                    if (updates.toString() == "[object FormData]") {
                        resetContentType();
                    }
                    dispatch(postSuccess());
                    resolve(result);
                })
                .catch((err) => {
                    resetContentType();
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * POST_FAIL.
 * action generator.
 * adds error to the store.
 *
 * @param {Object} error the error
 * @returns {Object} object to set error
 */
export const postFail = (error) => ({
    type: "POST_FAIL",
    error
});

/**
 * POST_SUCCESS.
 * action generator.
 * resets loading and error status.
 *
 * @returns {Object} object to reset loading and error status
 */
export const postSuccess = () => ({
    type: "POST_SUCCESS"
});

/**
 * CLEAR_POSTS.
 * action generator.
 * clears posts in store
 *
 * @returns {Object} object to reset posts
 */
export const clearPosts = () => ({
    type: "CLEAR_POSTS"
});

/**
 * searchPosts
 * action dispatcher - searches for posts
 * @param {string} text - text to search posts by
 * @returns {Promise} to be handled
 */
export const searchPosts = (text) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            API.get(`posts/?search=${text}`)
                .then((result) => {
                    dispatch(postSuccess());
                    dispatch(setPosts(result.data));
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * deletePost
 * action dispatcher - soft deletes a post
 * @param {number} id - id of post to delete
 * @param {Object} post - Post object to delete
 * @returns {Promise} to be handled
 */
export const deletePost = (id) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            const updates = {
                deleted_flag: true,
                deletion_date: moment().toDate()
            };
            API.patch(`posts/${id}/`, updates)
                .then((result) => {
                    dispatch(postSuccess());
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * restorePost
 * action dispatcher - restores a post
 * @param {number} id - id of post to restore
 * @param {Object} post - Post object to restore
 * @returns {Promise} to be handled
 */
export const restorePost = (id) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            const updates = {
                deleted_flag: false,
                deletion_date: null
            };
            API.patch(`posts/${id}/`, updates)
                .then((result) => {
                    dispatch(postSuccess());
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * getNonInterestPosts
 * action dispatcher - gets posts that are NOT relavent to your interests
 * @returns {Promise} to be handled
 */
export const getNonInterestPosts = () => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            API.get(`non-interest-posts/`)
                .then((result) => {
                    dispatch(postSuccess());
                    dispatch(setNonInterestPosts(result.data));
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));

                    reject(err);
                });
        });
    };
};

/**
 * setNonInterestPosts
 * action generator - sets posts not relavent to interests
 * @returns {Object} Action object
 */
export const setNonInterestPosts = (nonInterestPosts) => ({
    type: "SET_NON_INTEREST_POSTS",
    nonInterestPosts
});

/**
 * getRandomNonInterestPost
 * action dispatcher - Gets a random post that is not relavent to interests
 * @returns {Promise} Promise to handle
 */
export const getRandomNonInterestPost = () => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            API.get(`random-post/`)
                .then((result) => {
                    dispatch(postSuccess());
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};

/**
 * getCommunityPosts
 * action dispatcher - Gets posts only according to interests
 * @returns {Promise} Promise to handle
 */
export const getCommunityPosts = () => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            dispatch(postStart());
            API.get(`community-posts/`)
                .then((result) => {
                    dispatch(postSuccess());
                    dispatch(setPosts(result.data));
                    resolve(result);
                })
                .catch((err) => {
                    dispatch(postFail(err));
                    reject(err);
                });
        });
    };
};
