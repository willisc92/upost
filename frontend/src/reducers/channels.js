// Channels Reducers
const channelsDefaultState = [];

export default (state = channelsDefaultState, action) => {
    switch (action.type) {
        case "SET_CHANNELS":
            return {
                channels: action.channels
            };
        default:
            return state;
    }
};
