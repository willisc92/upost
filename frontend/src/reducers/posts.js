// Post Reducers
export const postsDefaultState = { error: null, loading: false, posts: [], interestRandomPosts: [] };

export default (state = postsDefaultState, action) => {
    switch (action.type) {
        case "POST_START":
            return {
                ...state,
                error: null,
                loading: true
            };
        case "SET_POSTS":
            return {
                ...state,
                posts: action.posts
            };
        case "POST_FAIL":
            return {
                ...state,
                error: action.error,
                loading: false
            };
        case "POST_SUCCESS":
            return {
                ...state,
                error: null,
                loading: false
            };
        case "SET_INTEREST_RANDOM_POSTS":
            return {
                ...state,
                interestRandomPosts: []
            };
        case "ADD_INTEREST_RANDOM_POSTS":
            return {
                ...state,
                interestRandomPosts: [...state.interestRandomPosts, action.interestPosts]
            };
        default:
            return state;
    }
};
