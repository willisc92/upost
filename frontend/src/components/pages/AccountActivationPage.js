import React from "react";
import { startActivation } from "../../actions/activate";

class AccountActivationPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            inProgress: true,
            isValid: undefined
        };
    }

    componentDidMount() {
        const uid = this.props.match.params.uid;
        const token = this.props.match.params.token;
        startActivation(uid, token)
            .then((result) => {
                this.setState(() => {
                    return { inProgress: false, isValid: true };
                });
            })
            .catch((error) => {
                console.log("error", JSON.stringify(error, null, 2));
                this.setState(() => {
                    return { inProgress: false, isValid: false };
                });
            });
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        {this.state.inProgress ? (
                            <h1 className="page-header__title">
                                Your <span>UPost</span> account is currently being activated!
                            </h1>
                        ) : this.state.isValid ? (
                            <h1 className="page-header__title">
                                Thank you for your email confirmation. Please login and complete your registration.
                            </h1>
                        ) : (
                            <h1 className="page-header__title">The activation link is invalid</h1>
                        )}
                    </div>
                </div>
            </div>
        );
    }
}

export default AccountActivationPage;
