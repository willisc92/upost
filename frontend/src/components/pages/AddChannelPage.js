import React from "react";
import ChannelForm from "../forms/ChannelForm";
import { connect } from "react-redux";
import { addChannel } from "../../actions/channels";

class AddChannelPage extends React.Component {
    constructor(props) {
        super(props);
    }

    onSubmit = (channel) => {
        this.props
            .addChannel({ ...channel })
            .then(() => this.props.history.push("/myChannels"))
            .catch(() => {});
    };

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add a Channel</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ChannelForm onSubmit={this.onSubmit} />
                </div>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    addChannel: (channel) => dispatch(addChannel(channel))
});

export default connect(
    null,
    mapDispatchToProps
)(AddChannelPage);
