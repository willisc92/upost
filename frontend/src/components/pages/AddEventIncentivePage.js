import React from "react";
import { addIncentivePackage } from "../../actions/incentivePackage";
import { getCurrentUser } from "../../actions/auth";
import IncentiveForm from "../forms/IncentiveForm";
import { connect } from "react-redux";
import { startSetEvent, clearEvents } from "../../actions/events";

import Box from "@material-ui/core/Box";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";

class AddEventIncentivePage extends React.Component {
    constructor(props) {
        super(props);
    }

    componentWillMount() {
        this.props.clearEvents();
        const event_id = this.props.match.params.id;
        getCurrentUser()
            .then((res) => {
                this.props
                    .startSetEvent(event_id)
                    .then((event_res) => {
                        if (res.data.username !== event_res.data.event_owner) {
                            this.props.history.push(`/myChannels`);
                        }
                    })
                    .catch((err) => {
                        console.log(JSON.stringify(err, null, 2));
                    });
            })
            .catch((err) => {
                console.log(JSON.stringify(err, null, 2));
            });
    }

    onSubmit = (incentive) => {
        const post_id = this.props.event.post;
        this.props
            .addIncentivePackage(incentive)
            .then((res) => this.props.history.push(`/post-events/${post_id}`))
            .catch((err) => {
                console.log(err);
            });
    };

    goBack = () => {
        const event_id = this.props.match.params.id;
        const post_id = this.props.event.post;
        this.props.history.push(`/myPosts/${post_id}/events/${event_id}/edit`);
    };

    goToIncentive = () => {
        const event_id = this.props.match.params.id;
        this.props.history.push(`/myEvents/${event_id}/editIncentive`);
    };

    render() {
        const read_only = !!this.props.event && this.props.event.deleted_flag;
        const existing_incentive = !!this.props.event && !!this.props.event.event_incentive;

        return (
            !!this.props.event && (
                <div>
                    <Box bgcolor="secondary.main" py={3}>
                        <Container fixed>
                            <Typography gutterBottom variant="h2">
                                Add an Incentive Package to Event:{" "}
                                <Typography variant="inherit" display="inline" color="primary">
                                    {this.props.event && this.props.event.event_title}
                                </Typography>
                            </Typography>

                            {existing_incentive ? (
                                <Box paddingBottom={2}>
                                    <Typography variant="h2" color="error" gutterBottom>
                                        There is already an existing incentive tied to this event.
                                    </Typography>
                                    <Button variant="contained" color="primary" onClick={this.goToIncentive}>
                                        Go to Incentive
                                    </Button>
                                </Box>
                            ) : (
                                read_only && (
                                    <Typography variant="h2" color="error" gutterBottom>
                                        The event this incentive will be tied to is deleted. Restore it before adding an
                                        incentive.
                                    </Typography>
                                )
                            )}
                            <Box>
                                <Button variant="contained" color="primary" onClick={this.goBack}>
                                    Go to Event
                                </Button>
                            </Box>
                        </Container>
                    </Box>
                    <Container fixed>
                        <IncentiveForm
                            onSubmit={this.onSubmit}
                            nextStep="Save"
                            event={this.props.event}
                            fromEvent={true}
                            read_only={read_only || existing_incentive}
                        />
                    </Container>
                </div>
            )
        );
    }
}

const mapStateToProps = (state) => ({
    event: state.events.events.length !== 0 ? state.events.events : null
});

const mapDispatchToProps = (dispatch) => ({
    clearEvents: () => dispatch(clearEvents()),
    addIncentivePackage: (incentive) => dispatch(addIncentivePackage(incentive)),
    startSetEvent: (id) => dispatch(startSetEvent(id))
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AddEventIncentivePage);
