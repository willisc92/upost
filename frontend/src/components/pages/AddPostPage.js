import React from "react";
import PostForm from "../forms/PostForm";
import { connect } from "react-redux";
import { addPost } from "../../actions/posts";
import { startGetChannel } from "../../actions/channels";
import { getCurrentUser } from "../../actions/auth";

export class AddPostPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            step: "Post",
            error: ""
        };
    }

    componentDidMount() {
        const channel_id = this.props.match.params.id;
        getCurrentUser()
            .then((res) => {
                this.props
                    .startGetChannel(channel_id)
                    .then((channel_res) => {
                        if (res.data.username !== channel_res.data[0].user) {
                            this.props.history.push("/myChannels");
                        }
                    })
                    .catch((err) => {
                        console.log(JSON.stringify(err, null, 2));
                    });
            })
            .catch((err) => {
                console.log(JSON.stringify(err, null, 2));
            });
    }

    onSubmit = (data) => {
        this.props
            .addPost(data)
            .then((result) => {
                if (this.state.step === "Post") {
                    this.handleReturn();
                } else if (this.state.step === "Event") {
                    this.props.history.push(`/myPosts/${result.data.post_id}/addEvent`);
                } else if (this.state.step === "Incentive") {
                    this.props.history.push(`/myPosts/${result.data.post_id}/addIncentive`);
                }
            })
            .catch((err) => {
                console.log(JSON.stringify(err, null, 2));
            });
    };

    handleReturn = () => {
        const channel_id = this.props.match.params.id;
        this.props.history.push(`/myChannels/${channel_id}`);
    };

    onTriggerSaveAddEvent = async () => {
        await this.setState(() => ({ step: "Event" }));
        this.submitButtonRef.click();
    };

    onTriggerSaveAddIncentive = async () => {
        await this.setState(() => ({ step: "Incentive" }));
        this.submitButtonRef.click();
    };

    render() {
        const read_only = !!this.props.channel && this.props.channel.deleted_flag;

        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">
                            Add Post to Channel: <span>{this.props.channel && this.props.channel.channel_name}</span>
                        </h1>
                        {read_only ? (
                            <div>
                                <h2 className="page-header__subtitle__red">
                                    You must restore the Channel of this post before adding.
                                </h2>
                                <button className="button" onClick={this.handleReturn}>
                                    Go to Channel
                                </button>
                            </div>
                        ) : (
                            <button className="button" onClick={this.handleReturn}>
                                Go Back
                            </button>
                        )}
                    </div>
                </div>
                <div className="content-container">
                    {!!this.state.error && <p className="form__error">{this.state.error}</p>}
                    <PostForm
                        id="Post"
                        onSubmit={this.onSubmit}
                        channel={this.props.match.params.id}
                        nextStep="Save Post and Return"
                        read_only={read_only}
                    />
                    {!read_only && (
                        <div className="input-group">
                            <div className="input-group__item">
                                <button className="button" onClick={this.onTriggerSaveAddEvent}>
                                    Save Post and Add Event
                                </button>
                            </div>
                            <div className="input-group__item">
                                <button className="button" onClick={this.onTriggerSaveAddIncentive}>
                                    Save Post and Add Incentive
                                </button>
                            </div>
                        </div>
                    )}
                    <button
                        className="button__invisible"
                        type="submit"
                        form="Post"
                        ref={(node) => {
                            this.submitButtonRef = node;
                        }}
                    />
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    channel: state.channels.channels[0]
});

const mapDispatchToProps = (dispatch) => ({
    addPost: (post) => dispatch(addPost(post)),
    startGetChannel: (channel_id) => dispatch(startGetChannel(channel_id))
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AddPostPage);
