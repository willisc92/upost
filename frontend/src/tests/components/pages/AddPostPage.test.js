import React from "react";
import { shallow } from "enzyme";
import { AddPostPage } from "../../../components/pages/AddPostPage";
import channels from "../../fixtures/channels";
import posts from "../../fixtures/posts";
import { writeHeapSnapshot } from "v8";

let wrapper, match, channel, addPost, startGetChannel, history;

beforeEach(() => {
    channel = channels[0];
    match = {
        params: {
            id: channel.channel_id
        }
    };

    history = {
        push: jest.fn()
    };

    addPost = jest.fn().mockImplementation(() => {
        return Promise.resolve();
    });

    startGetChannel = jest.fn().mockImplementation(() => {
        return Promise.resolve();
    });

    localStorage.setItem("user_name", channel.user);

    wrapper = shallow(
        <AddPostPage
            match={match}
            channel={channel}
            addPost={addPost}
            startGetChannel={startGetChannel}
            history={history}
        />
    );
});

test("Should render page with Add Post form correctly upon page load", () => {
    expect(wrapper).toMatchSnapshot();
});

test("Should return to channels page", () => {
    wrapper.instance().handleReturn();
    expect(history.push).toHaveBeenCalled();
    expect(history.push).toHaveBeenCalledWith(`/myChannels/${match.params.id}`);
});

test("Should handle triggering final submit", async () => {
    wrapper.instance().onTriggerSaveReturn();
    expect(wrapper.state("finished")).toBe(true);
});

test("Should handle triggering final submit", async () => {
    wrapper.instance().onTriggerSaveReturn();
    expect(wrapper.state("finished")).toBe(true);
});

test("Should handle intermediate submit and increment the step to Event", async () => {
    await wrapper.instance().onSubmit(posts[0]);
    expect(addPost).toHaveBeenCalledWith({ ...posts[0] });
    expect(wrapper.state("step")).toBe("Event");
    expect(wrapper).toMatchSnapshot();
});
