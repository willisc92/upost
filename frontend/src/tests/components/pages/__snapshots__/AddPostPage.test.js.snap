// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should handle intermediate submit and increment the step to Event 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Add a Post
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <Connect(PostForm)
      channel={1}
      id="Post"
      nextStep="Save Post and Add Event"
      onSubmit={[Function]}
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Save Post and Return to Channel
    </button>
    <button
      className="button__invisible"
      form="Post"
      type="submit"
    />
  </div>
</div>
`;

exports[`Should render page with Add Post form correctly upon page load 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Add a Post
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <Connect(PostForm)
      channel={1}
      id="Post"
      nextStep="Save Post and Add Event"
      onSubmit={[Function]}
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Save Post and Return to Channel
    </button>
    <button
      className="button__invisible"
      form="Post"
      type="submit"
    />
  </div>
</div>
`;
